kernelci/cli/user.py

```diff
@@ -0,0 +1,109 @@
+# SPDX-License-Identifier: LGPL-2.1-or-later
+#
+# Copyright (C) 2023 Collabora Limited
+# Author: Guillaume Tucker <guillaume.tucker@collabora.com>
+# Author: Jeny Sadadia <jeny.sadadia@collabora.com>
+# Author: Paweł Wieczorek <pawiecz@collabora.com>
+
+"""Tool to manage KernelCI API users"""
+
+import getpass
+import json
+
+import click
+
+import kernelci.api
+import kernelci.config
+from . import Args, kci, split_attributes
+
+
+@kci.group(name='user')
+def kci_user():
+    """Manage user accounts"""
+
+
+@kci_user.command(secrets=True)
+@Args.config
+@Args.api
+@Args.indent
+def whoami(config, api, indent, secrets):
+    """Get the current user's details with API authentication"""
+    configs = kernelci.config.load(config)
+    api_config = configs['api'][api]
+    api = kernelci.api.get_api(api_config, secrets.api.token)
+    data = api.whoami()
+    click.echo(json.dumps(data, indent=indent))
+
+
+@kci_user.command
+@click.argument('attributes', nargs=-1)
+@Args.config
+@Args.api
+@Args.indent
+def find(attributes, config, api, indent):
+    """Find user profiles with arbitrary attributes"""
+    configs = kernelci.config.load(config)
+    api_config = configs['api'][api]
+    api = kernelci.api.get_api(api_config)
+    users = api.get_user_profiles(split_attributes(attributes))
+    data = json.dumps(users, indent=indent)
+    echo = click.echo_via_pager if len(users) > 1 else click.echo
+    echo(data)
+
+
+@kci_user.command
+@click.argument('username')
+@Args.config
+@Args.api
+@Args.indent
+@click.option('--scope', multiple=True, help="Security scope(s)")
+def token(username, config, api, indent, scope):
+    """Create a new API token using a user name and password"""
+    password = getpass.getpass()
+    configs = kernelci.config.load(config)
+    api_config = configs['api'][api]
+    api = kernelci.api.get_api(api_config)
+    user_token = api.create_token(username, password, scope)
+    click.echo(json.dumps(user_token, indent=indent))
+
+
+@kci_user.group(name='password')
+def user_password():
+    """Manage user passwords"""
```

Comment:
After I added sub-commands to this group, it became `kci user password reset-password`.
I expected it to be `kci password reset-password`.
What are the benefits of adding groups?

Reply:
Well this was as per the proposed syntax on the GitHub discussion:
https://github.com/kernelci/kernelci-project/discussions/263#discussioncomment-7163554

It means all the user-related commands are grouped under `kci user`.  There might be other passwords for other things, like `kci storage password` or anything else.